src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/boss_grand_champions.cpp

```diff
@@ -80,43 +84,120 @@ enum Seats
     SEAT_ID_0                       = 0
 };
 
-/*
-struct Point
-{
-    float x, y, z;
-};
+Position const vehicleExitPos = { 759.680f, 665.237f, 412.392f };
 
-const Point MovementPoint[] =
+void EnterAggressiveMode(Creature* temp)
 {
-  {746.84f, 623.15f, 411.41f},
-  {747.96f, 620.29f, 411.09f},
-  {750.23f, 618.35f, 411.09f}
-};
-*/
-void AggroAllPlayers(Creature* temp)
+    temp->RemoveFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_NON_ATTACKABLE | UNIT_FLAG_IMMUNE_TO_PC);
+    temp->SetReactState(REACT_AGGRESSIVE);
+}
+
+void DoCastPennant(Creature* me)
```

Comment:
I needed to left 5 methods outside of toc_bossAI as they are also used by the lesser champions and Grand Champions' vehicles.